addons/point_of_sale/static/src/app/screens/payment_screen/payment_screen.js

```diff
@@ -468,25 +468,17 @@ export class PaymentScreen extends Component {
             line.can_be_reversed = false;
         });
 
-        const payment_terminal = line.payment_method.payment_terminal;
-        line.set_payment_status("waiting");
-
-        const isPaymentSuccessful = await payment_terminal.send_payment_request(line.cid);
-        if (isPaymentSuccessful) {
-            line.set_payment_status("done");
-            line.can_be_reversed = payment_terminal.supports_reversals;
-            // Automatically validate the order when after an electronic payment,
-            // the current order is fully paid and due is zero.
-            const { config, currency } = this.pos;
-            if (
-                this.currentOrder.is_paid() &&
-                floatIsZero(this.currentOrder.get_due(), currency.decimal_places) &&
-                config.auto_validate_terminal_payment
-            ) {
-                this.validateOrder(false);
-            }
-        } else {
-            line.set_payment_status("retry");
+        await line.pay();
+        // Automatically validate the order when after an electronic payment,
+        // the current order is fully paid and due is zero.
+        const { config, currency } = this.pos;
+        const currentOrder = this.pos.get_order();
+        if (
+            currentOrder.is_paid() &&
+            floatIsZero(currentOrder.get_due(), currency.decimal_places) &&
+            config.auto_validate_terminal_payment
+        ) {
+            this.validateOrder(false);
         }
```

Comment:
This should only be triggered when the payment is successful. Make the method `pay` return a boolean and use that to conditionally run this code.